src/main/scala/com/fulcrumgenomics/umi/ConsensusCallingIterator.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.util.ProgressLogger
+import htsjdk.samtools.SAMRecord
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/**
+  * An iterator that consumes from an incoming queue of SAMRecords and generates consensus
+  * read SAMRecords using the supplied caller.
+  *
+  * @param iterator an iterator of SAMRecords
+  * @param caller   the consensus caller to use to call consensus reads
+  * @param progress an optional progress logger to which to log progress in input reads
+  */
+class ConsensusCallingIterator(iterator: Iterator[SAMRecord],
+                               val caller: UmiConsensusCaller[_],
```

Comment:
I _think_ that since the iterator never sees any of the `ConsensusRead` objects and never has to write the type anywhere, it doesn't care.  I.e. I read this as `I the iterator need a consensus caller, you might be constrained on it's type parameter, but personally I don't care`.